main.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+from subprocess import check_output
+
+from slack_bolt.adapter.socket_mode import SocketModeHandler
+
+from shared import *
+
+for feat in config['features_enabled']:
+	exec(f"import {feat}")
+
+print("Starting parrotbot...", flush=True, file=info_stream)
+
+@app.command("/parrotcheckhealth")
+def parrotcheckhealth(client, ack, body, say):
+	MAX_CHARS=2500
+
+	log = check_output(config['log']['command'], shell=True, text=True).replace('files.slack.com', '********')
+	log_lines = log.split('\n')
+	log_messages = []
+
+	# incredibly jank but whatever
+	while len(log_lines) > 0:
+		msg = ""
+		while len(log_lines) > 0 and len(msg) + len(log_lines[0]) <= MAX_CHARS:
+			msg += log_lines.pop(0) + '\n'
+		log_messages.append(msg)
+
+	if "quiet" in body['text']:
+		ack("I'm running! Here is my latest log:" \
+			"\n```\n" + log_messages[-1] + '```')
+	else:
+		say("I'm running! Here is my log:")
+		for msg in log_messages:
+			say("\n```\n" + msg + '```', unfurl_media = False, unfurl_links=False)
+		ack()
+
+####### TODO: THIS ISN'T WORKING
+# @app.event("channel_created")
+# def handle_file_shared(client, event, ack):
+	# print('hello there', flush=True)
+	# if not chan['is_im']:
+		# chan=event['channel']
+		# app.client.conversations_join(channel=chan['id'])
+		# print(f"{datetime.now().isoformat()}: Joined {chan['name']}",
+				# flush=True)
+	# ack()
+
```

Comment:
This should be removed as `live_archive.py` handles joining new channels anyway.

Reply:
Agreed